Tests/FluentMySQLDriverTests/FluentMySQLDriverTests.swift

```diff
@@ -165,6 +165,97 @@ final class FluentMySQLDriverTests: XCTestCase {
         }
     }
 
+    func testBoolFilter() throws {
+        final class Clarity: Model {
+            static let schema = "clarities"
+
+            @ID(custom: .id, generatedBy: .database)
+            var id: Int?
+
+            @Field(key: "rain")
+            var rain: Bool
+
+            init() { }
+
+            init(rain: Bool) {
+                self.rain = rain
+            }
+        }
+
+        struct CreateClarity: Migration {
+            func prepare(on database: Database) -> EventLoopFuture<Void> {
+                return database.schema("clarities")
+                    .field("id", .int, .identifier(auto: true))
+                    .field("rain", .bool, .required)
+                    .create()
+            }
+
+            func revert(on database: Database) -> EventLoopFuture<Void> {
+                return database.schema("clarities").delete()
+            }
+        }
+
+        defer { try? CreateClarity().revert(on: self.db).wait() }
+        try CreateClarity().prepare(on: self.db).wait()
+
+        let trueValue = Clarity(rain: true)
+        let falseValue = Clarity(rain: false)
+
+        try trueValue.save(on: self.db).wait()
+        try falseValue.save(on: self.db).wait()
+
+        XCTAssertEqual(try Clarity.query(on: self.db).count().wait(), 2)
+        XCTAssertEqual(try Clarity.query(on: self.db).filter(\.$rain == true).first().wait()?.id, trueValue.id)
+        XCTAssertEqual(try Clarity.query(on: self.db).filter(\.$rain == false).first().wait()?.id, falseValue.id)
+    }
+
+    func testDateDecoding() throws {
+        final class Clarity: Model {
+            static let schema = "clarities"
+
+            @ID(custom: .id, generatedBy: .database)
+            var id: Int?
+
+            @Field(key: "date")
+            var date: Date
+
+            init() { }
+
+            init(date: Date) {
+                self.date = date
+            }
+        }
+
+        struct CreateClarity: Migration {
+            func prepare(on database: Database) -> EventLoopFuture<Void> {
+                return database.schema("clarities")
+                    .field("id", .int, .identifier(auto: true))
+                    .field("date", .date, .required)
```

Comment:
Ah, this one has already been reported a few times. I think what you want is `.datetime` here which works. Unless you truly want to store just the date without time. That said, obviously using `.date` should either work or give a useful error. I'm not sure which of those to do, though. Neither of these options are ideal:
- Silently drop time information 
- Require a separate type like `DateWithoutTime` or `GregorianDate`

Reply:
I was indeed hoping for `date` and not `datetime` but I suppose switching to `datetime` wouldn't be the end of the world.

Reply:
Actually, I take that back, I definitely need `date` functionality (connecting to legacy database structure that can't be updated)

Reply:
I see. We need to find a good solution for that. Were you just using `Date` in Fluent 3? If that worked by silently dropping time information then we could probably just stick to that for now. That's not ideal but if it's existing behavior we can worry about that later.

Reply:
Yeah it was just `Date`

Reply:
Fixed this by allowing Foundation.Date to convert to/from DATE by ignoring the time components.